### PR TITLE
Add nanolib to nng & fix build fail on Windows.

### PR DESCRIPTION
**Pull request by alvin1221**

### CODE REVIEW

File: include/nng/supplemental/nanolib/mqtt_db.h [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#ifndef MQTT_DB_H
+#define MQTT_DB_H
+
+#include "cvector.h"
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+typedef enum {
+	MQTT_VERSION_V311 = 4,
```
Comment (JaylinYu) [inline include/nng/supplemental/nanolib/mqtt_db.h:10]: This is redundant

File: src/sp/protocol/mqtt/mqtt_parser.c [hunk 2]
```diff
@@ -1696,11 +1700,14 @@ topic_filter(const char *origin, const char *input)
 bool
 topic_filtern(const char *origin, const char *input, size_t n)
 {
-	char buff[n+1];
-	memset(buff, '\0', n+1);
+	char *buff = nni_zalloc(n + 1);
```
Comment (JaylinYu) [inline src/sp/protocol/mqtt/mqtt_parser.c:1703]: better keep it as it was

**Review by JaylinYu — APPROVED** [review]

File: include/nng/supplemental/nanolib/mqtt_db.h (hunk 1 above)
Comment (alvin1221) [reply include/nng/supplemental/nanolib/mqtt_db.h:10]: I'll remove the enum.

File: src/sp/protocol/mqtt/mqtt_parser.c (hunk 2 above)
Comment (alvin1221) [reply src/sp/protocol/mqtt/mqtt_parser.c:1703]: compile fail on Windows msvc.

**Pull request merged**